dev-lang/rust_bin/rust_bin-1.29.0.recipe

```diff
@@ -0,0 +1,102 @@
+SUMMARY="Modern and safe systems programming language"
+DESCRIPTION="Rust is a systems programming language that runs blazingly fast, \
+prevents almost all crashes*, and eliminates data races."
+HOMEPAGE="https://www.rust-lang.org/"
+COPYRIGHT="2018 The Rust Project Developers"
+LICENSE="MIT"
+REVISION="1"
+
+case "$effectiveTargetArchitecture" in
+	x86)
+SOURCE_URI="http://dl.rust-on-haiku.com/dist/$portVersion/rust-$portVersion-i686-unknown-haiku.tar.xz"
+CHECKSUM_SHA256="b4d40d82bb6c07746b1ca8e407a23e45dd357744ee80217446bc9f25cc5e9eaa"
+SOURCE_DIR="rust-$portVersion-i686-unknown-haiku"
+		;;
+	x86_64)
+SOURCE_URI="http://dl.rust-on-haiku.com/dist/$portVersion/rust-$portVersion-x86_64-unknown-haiku.tar.xz"
+CHECKSUM_SHA256="e8229ee38453978485720e6a8ef84e0474aa5fb9375b3e74ecce655ee742b5f2"
+SOURCE_DIR="rust-$portVersion-x86_64-unknown-haiku"
+		;;
+	*)
+SOURCE_URI="http://dl.rust-on-haiku.com/dist/$portVersion/rustc-$portVersion-src.tar.xz"
+CHECKSUM_SHA256="cec7232a3081f7a29919523490b7045dd9637aee371de4051c4f9348b657f30e"
+SOURCE_DIR="rustc-$portVersion-src"
+		;;
+esac
+
+ARCHITECTURES="!x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+DISABLE_SOURCE_PACKAGE=yes
+
+cargoVersion="0.30.0"
+rlsVersion="0.130.0"
+rustfmtVersion="0.99.1"
+clippyVersion="0.0.212"
+
+PROVIDES="
+	rust_bin$secondaryArchSuffix = $portVersion
+	cmd:rustc = $portVersion
+	cmd:rustdoc = $portVersion
+	cmd:rust_gdb = $portVersion
+	cmd:rust_lldb = $portVersion
+	cmd:cargo$secondaryArchSuffix = $cargoVersion
+	cmd:cargo_clippy = $clippyVersion
+	cmd:cargo_fmt = $cargoVersion
+	cmd:clippy_driver = $clippyVersion
+	cmd:rls = $rlsVersion
+	cmd:rustfmt = $rustfmtVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libcrypto$secondaryArchSuffix
+	lib:libcurl$secondaryArchSuffix
+	lib:libssl$secondaryArchSuffix
+	lib:libssh2$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+CONFLICTS="
+	rust$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+
+INSTALL()
+{
+	./install.sh                                   \
+		--prefix=$prefix                           \
+		--docdir=$developDocDir                    \
+		--libdir=$libDir                           \
+		--mandir=$manDir                           \
+		--sysconfdir=$dataDir                      \
+		--disable-ldconfig
+
+	# move the cargo and binaries (in case of a secondary arch)
+	if [ -n "$secondaryArchSuffix" ]; then
+		mkdir -p $binDir
+		mv $prefix/bin/cargo $binDir/cargo
```

Comment:
May I know why this is necessary? And why is only `cargo` moved and not the others?

Reply:
I think this would only be useful if we wanted to allow cargo_x86 on x86_64.
If we don't need this then I think we can apply the suggestions and the "maybe" comments in my last review: https://github.com/haikuports/haikuports/pull/3147#pullrequestreview-160213323

Reply:
> I think this would only be useful if we wanted to allow cargo_x86 on x86_64.

And have `rust_x86` override `rust`? Sounds counterintuitive to me ;)

Reply:
Yeah, having the suffix only for `cmd:cargo` and not for the other `cmd:` runtimes is probably not a good idea, but I might be missing something.

Reply:
This was discussed in earlier PRs as well. The rustc binary depends on the libraries being in the relative subdir ../lib to the rustc binary. If we move rustc to bin/x86 this breaks and it can no longer find the libraries. 

Rustc does depend on the gcc 7+ toolchain. The main entrypoint to rust is the cargo binary. By putting the cargo binary behind the setarch x86 gate, we prevent that people accidentally call cargo in the gcc2 environment and create weird linking errors.

Reply:
OK. I _think_ I got it. (@alaviss, if you are OK with keeping the suffix for `cmd:cargo` then I guess we're almost done.)
I've just marked as resolved all the "maybe" comments I wrote earlier.

So now the only changes that would be nice to have are these 2 tiny changes:

```diff
+if [ "$targetArchitecture" = x86_gcc2 ]; then
 SECONDARY_ARCHITECTURES="x86"
+fi
```
(Yep, this one is for `rust_bin`. For `rust` it's `"?x86"` instead of `"x86"`.)